### PR TITLE
Vps 15 concurrent scene saving

### PR DESCRIPTION
**Pull request by rgin216**
## Issue

The existing save logic overwrote the entire scene on each save. When multiple instances of the editor were open, this caused race conditions where different instances would continuously overwrite each other’s changes. Additionally, only the first modified component would persist when multiple components were updated in a single session.

## Solution

Implemented a patch-based save system that only persists incremental changes instead of the full scene.

- Added a `PATCH /scene/:sceneId` endpoint and `patchScene` DAO logic to:
  - Update only modified components
  - Insert new components
  - Remove deleted components
  - Update scene-level fields independently  
- Introduced frontend patch generation (`getScenePatch`) to detect granular changes  
- Replaced full save calls with patch saves in the editor flow  
- Fixed frontend state issues by ensuring fresh data is fetched after save  
- Resolved multi-component update bug by ensuring all modified components are included and correctly persisted  
- Added comprehensive DAO tests covering updates, inserts, deletes, and mixed operations  

## Risk

- Patch logic increases complexity compared to full overwrite, which may introduce edge cases and race conditions if patch generation becomes inconsistent with backend expectations  
- Potential for duplicate components if insert/update conditions are not perfectly aligned  
- Existing flows that relied on full scene replacement may behave differently if not fully migrated to patch logic  

## Checklist

- [x] Acceptance criteria met  
- [ ] Wiki documentation is written and up to date  
- [x] Unit tests written and passing  
- [ ] Integration tests written and passing  
- [x] Continuous integration build passing  
(Not sure what to write in the wiki)

### CODE REVIEW

File: backend/src/routes/api/__tests__/scenarioApi.test.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ import auth from "../../../middleware/firebaseAuth.js";
 import scenarioAuth from "../../../middleware/scenarioAuth.js";
 import Access from "../../../db/models/access.js";
 
+jest.setTimeout(30000);
```
Comment (harbassan) [inline backend/src/routes/api/__tests__/scenarioApi.test.js]: 😮

File: frontend/src/features/authoring/SceneNavigator/ContextableThumb.tsx [hunk 2]
```diff
@@ -61,10 +62,20 @@ function ContextableThumb({
       .catch(handleGeneric);
   };
 
-  function switchScene(scene: Record<string, any>) {
+  async function switchScene(scene: Record<string, any>) {
     if (active) return;
 
-    saveScene(structuredClone(getScene()));
+    const patch = structuredClone(getScenePatch());
```
Comment (harbassan) [inline frontend/src/features/authoring/SceneNavigator/ContextableThumb.tsx]: you dont need to structure clone this, since youyr getScenePatch method is creating new objects

File: frontend/src/features/authoring/AuthoringToolPage.jsx [hunk 3]
```diff
@@ -82,9 +82,23 @@ export default function AuthoringToolPage() {
   async function save() {
     if (saving) return; // we dont want to interrupt in progress saves (usually uploading media)
     setSaving(true);
-    const clone = structuredClone(getScene());
-    await saveScene(clone);
-    setTimeout(() => setSaving(false), 5000); // debounce saves
+
+    const patch = structuredClone(getScenePatch());
```
Comment (harbassan) [inline frontend/src/features/authoring/AuthoringToolPage.jsx]: same here

**Review by harbassan — CHANGES_REQUESTED** [review]
and please remove the old save implementation from the codebase 😄

**Review by harbassan — APPROVED** [review]
absolutely lovely 😁

**Review by del-ereno — APPROVED** [review]
works well gjgj
